### PR TITLE
refactor!: use go-thousandeyes/v2

### PR DESCRIPTION
**Pull request by raul-te**
This contribution updates the codebase to use go-thousandeyes/v2. It includes changes for:

- using pointers for every `json`-tagged field
- using `bool` instead of `int` for binary-valued fields
- general documentation updates
- `gofmt`

Edit: I've attempted to fix the [workflow](https://github.com/thousandeyes/terraform-provider-thousandeyes/runs/6734732533) by bumping to 1.17 but it's still failing. I've noticed that it was [already failing before](https://github.com/thousandeyes/terraform-provider-thousandeyes/runs/6379066873), so I believe it's unrelated to  these changes. I'll fix it on a new pull request.

### CODE REVIEW

File: thousandeyes/schemas.go [hunk 1]
```diff
@@ -293,37 +294,37 @@ var schemas = map[string]*schema.Schema{
...
+										},
+									},
+								},
+							},
+						},
+					},
+				},
 				"notify_on_clear": {
 					Type:        schema.TypeInt,
 					Description: "set to 1 to trigger the notification when the alert clears.",
```
Comment (sfreitas-te) [inline thousandeyes/schemas.go:330]:
```suggestion
					Description: "Set to 1 to trigger the notification when the alert clears.",
```

**Review by sfreitas-te — APPROVED** [review]
Looks good

File: thousandeyes/data_source_thousandeyes_account_group.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func dataSourceThousandeyesAccountGroupRead(d *schema.ResourceData, meta interfa
 	var found *thousandeyes.SharedWithAccount
 
 	for _, account := range *accounts {
-		if account.AccountGroupName == searchName {
+		if *account.AccountGroupName == searchName {
```
Comment (xvilaca-te) [inline thousandeyes/data_source_thousandeyes_account_group.go:43]: As a suggestion, I believe that a getter that returns the value or empty if null makes this easier to read. Pointers can be error prone IMO.

**Review by xvilaca-te — APPROVED** [review]

File: thousandeyes/data_source_thousandeyes_account_group.go (hunk 2 above)
Comment (raul-te) [reply thousandeyes/data_source_thousandeyes_account_group.go:43]: That'd be a good improvement for `go-thousandeyes`. I'll take note of it.

**Pull request merged**